pkg/apis/validate/metadata.go

```diff
@@ -28,9 +29,9 @@ const MaxLength = 63
 func ObjectMetadata(meta metav1.Object) *apis.FieldError {
 	name := meta.GetName()
 
-	if strings.Contains(name, ".") {
+	if err := validation.IsDNS1123Subdomain(name); len(err) > 0 {
```

Comment:
thanks a bunch @psschwei for thorough analysis on this.

After validating the resource name to be a valid `DNSSubdomain`, I would like to add the length check back to make sure the name does not exceed 63 characters (magical number 🤣) just like before. But this time instead of hard coding the number, we can rely on a constant defined in API machinery `DNS1123LabelMaxLength`

Reply:
API machinery constant added to #L27

Reply:
Does this mean that we *allowed* name like `fooBar` and now we don't ? Or was it already not allowed by the kubernetes api server ?

Reply:
> Does this mean that we _allowed_ name like `fooBar` and now we don't ? Or was it already not allowed by the kubernetes api server ?

Answered this below in main comments, but camelCase names were not previously allowed (blocked by either the api server of knative validation package, so this involves no change in functionality